parcels/field.py

```diff
@@ -1635,8 +1635,91 @@ def spatial_c_grid_interpolation3D(self, ti, z, y, x, time, particle=None):
             w = self.W.units.to_target(w, x, y, z)
         return (u, v, w)
 
+    def _is_land2D(self, di, yi, xi):
+        if self.U.data.ndim == 3:
+            if di < np.shape(self.U.data)[0]:
+                return (self.U.data[di, yi, xi] == 0.) and (self.V.data[di, yi, xi] == 0.)
```

Comment:
"float == 0"-check error. Please replace with "np.isclose(0)".

Reply:
function signature:
```
numpy.isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False)
```

might need to set `equal_nan=True` in this case.

Reply:
Fixed now